public/js/actions/api.js

```diff
@@ -0,0 +1,30 @@
+import $ from 'jquery';
+
+import * as defaultSettings from 'settings';
+
+var csrfToken;
+
+
+export function fetch(request, {jquery=$, settings=defaultSettings} = {}) {
+  request = Object.assign({
+    dataType: 'json',
+    headers: {},
+    xhrFields: {withCredentials: true},
```

Comment:
The setting sends cookies on Access Control requests. I don't know what that fully means but it seems like maybe we shouldn't do that all the time. Good call. I changed it so it only does it when the app is running on a known host. I added some comments about it too.